tycho-surefire/tycho-surefire-plugin/src/main/java/org/eclipse/tycho/surefire/AbstractTestMojo.java

```diff
@@ -699,6 +701,9 @@ public void execute() throws MojoExecutionException, MojoFailureException {
             return;
         }
         if (shouldRun()) {
+            // Allow constructing the test runtime against filtered OSGi/PDE metadata
+            project.setContextValue(TychoConstants.CTX_METADATA_ARTIFACT_LOCATION, metadataDirectory);
```

Comment:
One should not use the project to store values, please always use ReactorProject for this

Reply:
@laeubi that's fine! I will have to go back and change the `MavenProject` usage in the other PR too then.

Reply:
You can jsut include all changes in this PR

Reply:
@laeubi done! Should be ok now.